cmd/cctool/main.go

```diff
@@ -14,6 +14,7 @@ import (
 var cleanup sync.WaitGroup
 
 type commonConfig struct {
+	cleanup *sync.WaitGroup
```

Comment:
removing the pointer means the zero-value of commonConfig is OK to use

Reply:
Is it okay to copy the wait group between method calls? I defaulted to the sync.Mutex rules where you dont want to copy the mutex

Reply:
the commonConfig is passed by a pointer to the subcommand, so it should be fine.

Reply:
hmm, imo the usage of the cleanup waitgroup shouldn't break by you deciding to pass a reference or copy of the containing struct to a method. that seems like a hard to catch bug?

Reply:
you'd have to purposefully copy the struct, as it comes in as a pointer

Reply:
Well I get that, but if you wrote a function that *didn't* to that, you'd break the usage of commonConfig, which is weird.